### DISCUSSION TITLE
Simulating 2d microstructure having notch geometry

### DISCUSSION THREAD

**Discussion by vivy234** [neperfepx/FEPX#28] [Q&A] [answered]
Hi,
I have a 2d microstructure with a notch geometry generated by neper. Can we deform uniaxially this microstructure using  FEPX solver? Please suggest me.   
Thank you.
Regards
Vivek
![img](https://user-images.githubusercontent.com/60852040/125277154-b59e8e00-e32e-11eb-8788-3a5e524137ed.png)

**Comment by acmelab-ua** [comment, marked as answer]
Hi @vivy234,

FEPX performs simulations only on three-dimensional domains, it does not handle two-dimensional problems.

**Reply by vivy234** [reply]
Thank you for your quick response. What will be your suggestion on the three-dimension domain having notch geometry? I tried to simulate it. The simulation shows an error ' 6 surfaces necessary in $Fasetsan'. Then, I converted $Fasets 8 into $Fasets 6 
 and merge the notch geometry fasets value (say cut1) and fasets y0-2 into fasets Y0_1 in .msh file generated by neper. Now, simulation works well but I do not know, is this procedure correct or not? Please enlighten your comment on this.     
![imgsh](https://user-images.githubusercontent.com/60852040/125316453-c31b3e80-e355-11eb-810e-f9c7cb4a5404.png)

**Reply by rquey** [reply]
@mkasemer may want to elaborate, but yes, technically, I think that this works.

**Reply by vivy234** [reply]
Thank you for your response. I think the attached pictures clear you more. In image1, $fasets 8 has been changed to $fasets 6 to avoid the error shown in the attached image. In the same $fasets group, I have also added the cut1 subsection and y0-2 subsection into a subsection of Y0_1. I hope now I am able to clear my doubt. 
![image1](https://user-images.githubusercontent.com/60852040/125337331-7e4ed200-e36c-11eb-9c38-7a3fccc85824.png)
![error](https://user-images.githubusercontent.com/60852040/125337350-84dd4980-e36c-11eb-96c4-344593a97104.png)

**Reply by acmelab-ua** [reply]
Essentially, FEPX uses the surfaces for boundary condition application as well as surface force calculation.

Currently, FEPX only natively considers domains that are rectangular prisms, with surfaces `x0`, `x1`, `y0`, `y1`, `z0`, `z1`. However, it doesn't know or care if the surfaces are orthonormal, or on a cuboidal domain. You can "trick" FEPX into recognizing a non-cuboidal domain by listing 2 surfaces repeated 3 times. For instance, if you want to run a uniaxial simulation on a cylindrical domain, you could simply repeat the top and bottom of the cylinder 3 times. Boundary conditions would then be properly applied to the two parallel surfaces. Note: we plan on eventually implementing native recognition of other domain types in a future release.

The domain that you show has 8 surfaces, hence why there are 8 listed in the mesh file. By simply reducing "8" to "6", you don't necessarily list the surfaces that you care most about---you're just telling FEPX to read in only the first 6 surfaces. My suggestion is to identify what surfaces are most important for you (I assume the "left" and the "right" surfaces in the above 3D plot, let's call them `x0` and `x1`, though it depends on your coordinate system) and modifying the mesh file such that you repeat these sets 3 times (`x0`, `x1`, `x0`, `x1`, `x0`, `x1`). This is the safest and most general---I cannot guarantee that a more complex domain may act differently since you may not capture the surfaces of interest.

Does this make sense?

**Reply by dikshanshs** [reply]
@acmelab-ua what if we merge the notch area and other two split y surfaces to just y, that is combine 3 surfaces to create one surface that would sum the Faset to six.

**Reply by acmelab-ua** [reply]
One answer: yes, this could potentially work, so long as you properly merge these surfaces and don't intend to use them as boundary condition control surfaces. However, the integrated forces for this surface wouldn't make much sense.

A more conservative suggestion: You should not merge surfaces---this is potentially dangerous (same reasons as above for just reducing "8" to "6"), and surfaces should always be kept separate. I would follow the suggestions of my previous comment.

As another note: I notice that there are some small elements near grain boundaries. I'm not sure if you turned on regularization when generating the mesh in Neper, but this will lead to a more robust mesh overall, and will facilitate better simulations.

**Comment by Anmol-Bakshi** [comment]
> Hi, I have a 2d microstructure with a notch geometry generated by neper. Can we deform uniaxially this microstructure using FEPX solver? Please suggest me. Thank you. Regards Vivek ![img](https://user-images.githubusercontent.com/60852040/125277154-b59e8e00-e32e-11eb-8788-3a5e524137ed.png)

Can u tell the commands by which u have created this 2d microstructure using Neper?
